### PR TITLE
Metadata changes for Template & existing connectors

### PR DESCRIPTION
**Pull request by v-rucdu**
Metadata changes for Template & existing connectors

## Proposed Changes

  -
  -
  -

### CODE REVIEW

**Review by nazang — APPROVED** [review]

**Review by preetikr — CHANGES_REQUESTED** [review]
In general please capitalize - for e.g. Kind: Community and Tier: "Developer" etc.

File: DataConnectors/Perimeter81ActivityLogs.json [hunk 1]
```diff
@@ -99,5 +99,21 @@
...
+    "version": "1.0",
+    "kind": "dataConnector",
+    "source": {
+        "kind": "community"
+    },
+    "author": {
+        "name": "Perimeter 81"
+    },
+    "support": {
+        "name": "Perimeter 82",
```
Comment (preetikr) [inline DataConnectors/Perimeter81ActivityLogs.json:114]: The support name doesn't seem to be correct - shouldn't it be Perimeter 81?

File: DataConnectors/Templates/Connector_CEF_template.json [hunk 2]
```diff
@@ -110,5 +110,23 @@
...
             "description": "Make sure to configure the machine's security according to your organization's security policy\n\n\n[Learn more >](https://aka.ms/SecureCEF)"
         }
-    ]
+    ],
+    "metadata": {
+        "id": "Unique Identifier (GUID) used to identify dependencies and content from solutions or community.",
+        "version": "Optional: Version of the content. Default and recommended format is numeric (e.g. 1, 1.0, 1.0.0, 1.0.0.0), following ARM template best practices.",
+        "kind": "dataConnector",
+        "source": {
+            "kind": "source type of the content. Value must be one of these : localWorkspace | community | solution | sourceRepository",
```
Comment (preetikr) [inline DataConnectors/Templates/Connector_CEF_template.json:119]: Need to capitalize these values or is that managed at UX  layer?

File: DataConnectors/Templates/Connector_CEF_template.json [hunk 3]
```diff
@@ -110,5 +110,23 @@
...
+            "name": "Name of the content source. The repo name, solution name, LA workspace name etc."
+        },
+        "author": {
+            "name": "Name of the author. For localWorkspace it is automatically the workspace user"
+        },
+        "support": {
+            "tier": "Type of support for content item: microsoft | developer | community",
```
Comment (preetikr) [inline DataConnectors/Templates/Connector_CEF_template.json:126]: Need to capitalize these values or is that managed at UX layer?

File: DataConnectors/Templates/Connector_REST_API_AzureFunctionApp_template/DataConnector_API_AzureFunctionApp_template.json [hunk 4]
```diff
@@ -124,5 +124,23 @@
...
         }
-    ]
+    ],
+    "metadata": {
+        "id": "Unique Identifier (GUID) used to identify dependencies and content from solutions or community.",
+        "version": "Optional: Version of the content. Default and recommended format is numeric (e.g. 1, 1.0, 1.0.0, 1.0.0.0), following ARM template best practices.",
```
Comment (preetikr) [inline DataConnectors/Templates/Connector_REST_API_AzureFunctionApp_template/DataConnector_API_AzureFunctionApp_template.json:130]: Regarding version modify as follows:
1.  For kind = "community" and "solutions" the value is 1.0.0 format aligning with solutions which makes it easier to manage content. 
2. For kind = "sourceRepository" value recommended format is numeric (e.g. 1, 1.0, 1.0.0, etc.) following ARM template best practices.

File: DataConnectors/Templates/Connector_REST_API_template.json [hunk 5]
```diff
@@ -89,5 +89,23 @@
                 }
             ]
         }
-    ]
+    ],
+    "metadata": {
+        "id": "Unique Identifier (GUID) used to identify dependencies and content from solutions or community.",
+        "version": "Optional: Version of the content. Default and recommended format is numeric (e.g. 1, 1.0, 1.0.0, 1.0.0.0), following ARM template best practices.",
```
Comment (preetikr) [inline DataConnectors/Templates/Connector_REST_API_template.json:95]: apply version feedback from above

File: DataConnectors/Templates/Connector_Syslog_template.json [hunk 6]
```diff
@@ -106,5 +106,23 @@
                 }
             ]
         }
-    ]
+    ],
+    "metadata": {
+        "id": "Unique Identifier (GUID) used to identify dependencies and content from solutions or community.",
+        "version": "Optional: Version of the content. Default and recommended format is numeric (e.g. 1, 1.0, 1.0.0, 1.0.0.0), following ARM template best practices.",
```
Comment (preetikr) [inline DataConnectors/Templates/Connector_Syslog_template.json:112]: apply version feedback from above

**Review by preetikr — CHANGES_REQUESTED** [review]

**Comment by preetikr** [comment]
@v-rucdu and @v-jayakal - please resolve conflicts and share inputs on the changes requested.

**Comment by v-rucdu** [comment]
> In general shouldn't we capitalize values for e.g. Kind: Community and Tier: "Developer" etc. ? Or are these supposed to be handled in the UX? Starting with capitalized values will enable easier management on UX layer, I think.

>> These values are from the enums defined in the schema [here](https://msazure.visualstudio.com/One/_git/Sentinel-AzMP-Solutions?path=%2Fschema%2FcontentMetadata.schema.json). Also, these fields are not shown on UI not atleast for Data Connectors.

File: DataConnectors/Perimeter81ActivityLogs.json (hunk 1 above)
Comment (v-rucdu) [reply DataConnectors/Perimeter81ActivityLogs.json:114]: The value is as mentioned in [Data Connector Support Metadata.xlsx](https://microsoft-my.sharepoint.com/:x:/r/personal/nyassin_microsoft_com/_layouts/15/guestaccess.aspx?e=FfUUgU&CID=F0FE2683-3E74-4584-A675-E62A83121C4E&wdLOR=c360F89BB-01B1-452A-82FB-E7E91CAF248B&share=EVpUqzObGe5BvFGzHwLLvDYBjx7XUFTUkl6jwXixWj2r8A)

File: DataConnectors/Templates/Connector_CEF_template.json (hunk 3 above)
Comment (v-rucdu) [reply DataConnectors/Templates/Connector_CEF_template.json:126]: These values are from the enum as mentioned in [contentMetadataSchema](https://msazure.visualstudio.com/One/_git/Sentinel-AzMP-Solutions?path=%2Fschema%2FcontentMetadata.schema.json)

File: DataConnectors/Templates/Connector_CEF_template.json (hunk 2 above)
Comment (v-rucdu) [reply DataConnectors/Templates/Connector_CEF_template.json:119]: These values are from the enums defined in the schema here. Also, these fields are not shown on UI not atleast for Data Connectors.

File: DataConnectors/Templates/Connector_REST_API_template.json (hunk 5 above)
Comment (v-rucdu) [reply DataConnectors/Templates/Connector_REST_API_template.json]: Done

**Comment by v-rucdu** [comment]
> @v-rucdu and @v-jayakal - please resolve conflicts and share inputs on the changes requested.

Done

File: DataConnectors/Perimeter81ActivityLogs.json (hunk 1 above)
Comment (preetikr) [reply DataConnectors/Perimeter81ActivityLogs.json:114]: This needs to be "Perimeter 81" - I think it's a typo in the excel - please fix for correct company name  - there's no Perimeter 82 company.

**Review by preetikr — CHANGES_REQUESTED** [review]
Please make the following changes:
1. Perimeter 81 support needs to be "Perimeter 81" -  please fix for correct company name  - there's no Perimeter 82 company. 
2. For the template updates for version - we need a better official text. I was explaining the process. So update to the following as the official text in all places:
"version": "Optional: Version of the content. 1. For kind = \"community\" and \"solutions\" the value is 1.0.0 format  2. For kind = \"sourceRepository\" value recommended format is numeric (e.g. 1, 1.0, 1.0.0, etc.) following ARM template best practices.","
3. Update all FROM "version": "1.0" TO "version": "1.0.0" - to align with the guidance mentioned in templates :)  Thanks.

**Comment by v-rucdu** [comment]
> Please make the following changes:
> 
> 1. Perimeter 81 support needs to be "Perimeter 81" -  please fix for correct company name  - there's no Perimeter 82 company.
> 2. For the template updates for version - we need a better official text. I was explaining the process. So update to the following as the official text in all places:
>    "version": "Optional: Version of the content. 1. For kind = "community" and "solutions" the value is 1.0.0 format  2. For kind = "sourceRepository" value recommended format is numeric (e.g. 1, 1.0, 1.0.0, etc.) following ARM template best practices.","
> 3. Update all FROM "version": "1.0" TO "version": "1.0.0" - to align with the guidance mentioned in templates :)  Thanks.

Done... Please have a look. Thanks!!!

**Review by preetikr — APPROVED** [review]

**Pull request merged**